src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py

```diff
@@ -1119,6 +1120,42 @@ def get_node_settings(_flow_dag_path: Path):
             )
             assert get_node_settings(Path(source)) != get_node_settings(new_flow_dag_path)
 
+    def test_flow_build_executable(self):
+        source = f"{FLOWS_DIR}/web_classification/flow.dag.yaml"
+        target = "promptflow._sdk.operations._flow_operations.FlowOperations._run_pyinstaller"
+        with mock.patch(target) as mocked:
+            mocked.return_value = None
+
+            with tempfile.TemporaryDirectory() as temp_dir:
+                run_pf_command(
+                    "flow",
+                    "build",
+                    "--source",
+                    source,
+                    "--output",
+                    temp_dir,
+                    "--format",
+                    "executable",
+                )
+                # Start the Python script as a subprocess
+                app_file = Path(temp_dir, "app.py").as_posix()
+                process = subprocess.Popen(['python', app_file])
+
+                try:
+                    # Wait for a specified time (in seconds)
+                    wait_time = 5
+                    process.wait(timeout=wait_time)
+                    if process.returncode == 0:
+                        pass
+                    else:
+                        raise Exception(f"Process terminated with exit code {process.returncode}")
```

Comment:
will the ci log contain the error message in case of fail?

Reply:
not for now ,will only show 
![Uploading image.png…]()

Reply:
will refine in the next pr

Reply:
update： 
![image](https://github.com/microsoft/promptflow/assets/26239730/9c985072-c30c-4bd9-960e-b3748f178c24)